src/plugins/data/public/ui/search_bar/lib/use_saved_query.ts

```diff
@@ -71,6 +71,7 @@ export const useSavedQuery = (props: UseSavedQueriesProps): UseSavedQueriesRetur
     };
 
     if (props.savedQueryId) fetchSavedQuery(props.savedQueryId);
+    else setSavedQuery(undefined);
```

Comment:
@lizozom when I've modified discover's URL, removed the `savedQuery` part, there was no change in the savedQuery popup. with this change, this works